packages/contracts-bedrock/src/deploy-config.ts

```diff
@@ -0,0 +1,339 @@
+import { ethers } from 'ethers'
+
+/**
+ * Core required deployment configuration.
+ */
+interface RequiredDeployConfig {
+  /**
+   * Address that will own the entire system on L1 when the deploy is complete.
+   */
+  finalSystemOwner: string
+
+  /**
+   * Address that will own the entire system on L1 during the deployment process. This address will
+   * not own the system after the deployment is complete, ownership will be transferred to the
+   * final system owner.
+   */
+  controller: string
+
+  /**
+   * The L2 genesis script uses this to fill the storage of the L1Block info predeploy. The rollup
+   * config script uses this to fill the L1 genesis info for the rollup. The Output oracle deploy
+   * script may use it if the L2 starting timestamp is undefined, assuming the L2 genesis is set up
+   * with this.
+   */
+  l1StartingBlockTag: string
+
+  /**
+   * Chain ID for the L1 network.
+   */
+  l1ChainID: number
+
+  /**
+   * Chain ID for the L2 network.
+   */
+  l2ChainID: number
+
+  /**
+   * Number of seconds in between each L2 block.
+   */
+  l2BlockTime: number
+
+  /**
+   * Sequencer batches may not be more than maxSequencerDrift seconds after the L1 timestamp of the
+   * end of the sequencing window end.
+   */
+  maxSequencerDrift: number
+
+  /**
+   * Number of L1 blocks per sequencing window.
+   */
+  sequencerWindowSize: number
+
+  /**
+   * Number of seconds (w.r.t. L1 time) that a frame can be valid when included in L1.
+   */
+  channelTimeout: number
+
+  /**
+   * Address of the key the sequencer uses to sign blocks on the P2P layer.
+   */
+  p2pSequencerAddress: string
+
+  /**
+   * L1 address that batches are sent to.
+   */
+  batchInboxAddress: string
+
+  /**
+   * Acceptable batch-sender address, to filter transactions going into the batchInboxAddress on L1 for data.
+   * Warning: this address is hardcoded now, but is intended to become governed via L1.
```

Comment:
I believe this is the initial batch sender address, used for deployment of the `SystemConfig`